temporal-sdk/src/main/java/io/temporal/common/converter/DefaultDataConverter.java

```diff
@@ -119,6 +134,14 @@ public DefaultDataConverter withPayloadConverterOverrides(
     return this;
   }
 
+  /** Modifies this {@code DefaultDataConverter} by overriding its {@link FailureConverter}. */
+  public DefaultDataConverter withFailureConverterOverride(
```

Comment:
Let's make it just `withFailureConverter`. In PayloadConverter that `Override` word was added to show that the passed converters with not replace the full collection, but only the relevant "overrode"e elements. I don't see any reason to add it here, it will only make the comprehension of this method harder.

Reply:
> I think we should also add a VERY load and clear doc that most users don't need this method. And something like "If you think about encoding the Failures, see CodecPayloadConverter". Users SHOULDN'T typically use this method for failure encoding.

Reply:
Agree. I can't think of a good reason anyone needs a failure converter and since its inception a few months ago I haven't heard of one use of it in the wild (which is why we've been fine for so long not exposing failure conversion semantics to users until now).